spec.md

```diff
@@ -406,18 +404,28 @@ See
 [CloudEvent Attributes Extensions](primer.md#cloudevent-attribute-extensions)
 for additional information concerning the use and definition of extensions.
 
-This specification places no restriction on the type or semantics of the
-extension attributes. Each definition of an extensions SHOULD fully define all
-aspects of the attribute - e.g. its name, semantic meaning and possible values
-or even to indicate that it places no restrictions on its values. New extension
-definitions SHOULD use a name that is descriptive enough to reduce the chances
-of name collisions with other extensions. In particular, extension authors
-SHOULD check the [documented extensions](documented-extensions.md) document for
-the set of known extensions - not just for possible name conflicts but for
-extensions that might be of interest.
-
-Each specification that defines how to serialize a CloudEvent will define how
-extension attributes will appear.
+Extension attributes MUST conform to one of the types in the [Type
+System](#type-system).  This specification does not otherwise constrain the
+content or semantics of extension attributes.
+
+The definition of an extension attribute SHOULD define its [type](#type-system),
+name, semantic meaning and constraints on legal values.
+
+New extension definitions SHOULD use a name that is descriptive enough to reduce
+the chances of name collisions with other extensions. In particular, extension
+authors SHOULD check the [documented extensions](documented-extensions.md)
+document for the set of known extensions - not just for possible name conflicts
+but for extensions that might be of interest.
+
+The consumer of an extension attribute MUST know the extension definition to use
+it meaningfully, including being able to interpret the [canonical
+string-encoding](#type-system) for the attributes type.
+
+Intermediaries that need to transcode events between different formats or
+transport bindings do not need to understand the extension semantics. They can
+forward extension values either by mapping between the native types of each
+format, or by converting to the [canonical string-encoding](#type-system). The
+final consumer is responsible for determining the semantics of the attribute.
```

Comment:
Seems ok except the last sentence since technically an intermediary could understand and do something with an extension too, if it wanted.  Could we remove it?  Or perhaps say.... "In those cases, the final consumer....".  Then it's clear that it's just about situations where the intermediaries are unaware of the extension.